datalad/distribution/dataset.py

```diff
@@ -451,6 +385,52 @@ def subds_contains_path(ds, path):
 
         return Dataset(sds_path)
 
+    # TODO alternative: make them semi-commands in api/coreapi
+    @borrowdoc(GitRepo)
+    def call_git(self, args, files=None,
+                 expect_stderr=False, expect_fail=False, read_only=False):
+
+        return self._vcs.call_git(args, files=None, expect_stderr=False,
+                                  expect_fail=False, read_only=False)
+
+    @borrowdoc(GitRepo)
+    def call_git_items_(self, args, files=None, expect_stderr=False, sep=None,
+                        read_only=False):
+        return self._vcs.call_git_items_(args, files=None, expect_stderr=False,
+                                         sep=None, read_only=False)
+
+    @borrowdoc(GitRepo)
+    def call_git_oneline(self, args, files=None, expect_stderr=False,
+                         read_only=False):
+        return self._vcs.call_git_oneline(args, files=None, expect_stderr=False,
+                                          read_only=False)
+
+    @borrowdoc(GitRepo)
+    def call_git_success(self, args, files=None,
+                         expect_stderr=False, read_only=False):
+        return self._vcs.call_git_success(args, files=None, expect_stderr=False,
+                                          read_only=False)
+
+    @borrowdoc(AnnexRepo)
+    def call_annex_records(self, args, files=None):
+
+        return self._vcs.call_annex_records(args, files=None)
+
+    @borrowdoc(AnnexRepo)
+    def call_annex(self, args, files=None):
+
+        return self._vcs.call_annex(args, files=None)
+
+    @borrowdoc(AnnexRepo)
+    def call_annex_items_(self, args, files=None, sep=None):
+
+        return self._vcs.call_annex_items_(args, files=None, sep=None)
+
+    @borrowdoc(AnnexRepo)
+    def call_annex_oneline(self, args, files=None):
+
+        return self._vcs.call_annex_oneline(args, files=None)
```

Comment:
@mih said:

> All call_annex_*() methods would fail if the underlying VCS instance points to a GitRepo.

Reply:
Yes, this draft so far is meant to illustrate what I wrote in the issue in principle, not be a full implementation yet. I'd like to first get some agreement on the goal. However - yes: At the Dataset level, those would have to be properly guarded.